lib/src/api.dart

```diff
@@ -37,19 +37,16 @@ abstract class Serializer {
   static Object fromJson(String json, [Type type]) => _fromJson(json, type);
 
   /// Convert a Map<String, dynamic>
-  static Object fromMap(Map map, [Type type]) => _fromMap(map, type);
+  static Object fromMap(Map map, [Type type, List<Type> mapOf]) => _fromMap(map, type, mapOf);
```

Comment:
maybe `Type mapKey` and `Type mapValue` instead of a list, except if you see an use case with more that 2 type in the list, and are you using `mapOf[0]` ?